libpromises/cf3parse_logic.h

```diff
@@ -967,6 +985,11 @@ static inline void ParserHandleBlockAttributeRval()
                 break;
             }
         }
+
+        if (body_syntax->status == SYNTAX_STATUS_CUSTOM)
+        {
+            free(constraint_syntax);
+        }
```

Comment:
This is okay, but be careful with this pattern. It easily becomes unmaintainable and the next person will be likely to introduce a bug.

Reply:
Struggled finding another way to solve this, without rewriting too much code